.github/workflows/lint.yml

```diff
@@ -0,0 +1,47 @@
+name: Linting
+
+on:
+  push:
+    paths:
+      - '**.py'
+  pull_request:
+    paths:
+      - '**.py'
+
+jobs:
+  Black:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: psf/black@stable
+        with:
+          options: "--check --verbose"
+          src: "./src ./tests"
+  Ruff:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+      - name: Install ruff
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install ruff
+      - name: Run ruff
+        run: |
+          ruff src
+  MyPy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+      - name: Install Project and MyPy
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install .[lint]
```

Comment:
Add black and ruff to `lint` as well?

Reply:
They're already there, but I was trying to avoid the full install in the Black and Ruff jobs to keep things running faster. I only did it here as it also grabs all of the project dependencies, which are needed for MyPy to run correctly.